### ISSUE TITLE
Remove die function

### ISSUE THREAD

**Issue by mvdmakesthings** [DBSInteractive/DBS-Share-Count#3]
Die function kills the rest of the page when a problem with one of the social APIs happens. 

``` php
// Don't Do This...
if( curl_error( $ch ) ) {
    die( curl_error($ch) );
 }
```

**Comment by mvdmakesthings** [comment]
Want to take a look at this?

**Comment by dbswebsite** [comment]
Never have a die() like that in the page flow for http (you can do it with
ajax requests).

Something like

```
     if( curl_error( $ch ) || empty( $cont ) ) {
        error_log( 'Share Count: ' . curl_error($ch) );
    }
```

The extra empty() statement is if curl itself or the network connection
barfs and returns an empty string or somesuch.

I would do the same for any curl statements (here or anywhere else).

On Tue, Oct 20, 2015 at 11:40 AM, Michael Large notifications@github.com
wrote:

> Want to take a look at this?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/DBSInteractive/DBS-Share-Count/issues/3#issuecomment-149608382
> .

## 

Hal Burgiss
CTO and DevOps
_DBS__>Interactive_ http://www.dbswebsite.com/

## 

NOTICE: This E-mail (including attachments) is covered by the Electronic Communications 
Privacy Act, 18 U.S.C. 2510-2521, is confidential and may be legally 
privileged. If you are not the intended recipient, you are hereby notified 
that any retention, dissemination, distribution or copying of this communication 
is prohibited. Please reply to the sender that you have received the 
message in error, then delete it.

**Comment by dbswebsite** [comment]
Should handle the return differently too:

  if( curl_error( $ch ) || empty( $cont ) ) {
            error_log( 'Share Count: ' . curl_error($ch) );
            return '';
   }

On Tue, Oct 20, 2015 at 12:00 PM, Hal Burgiss hal@dbsinteractive.com
wrote:

> Never have a die() like that in the page flow for http (you can do it with
> ajax requests).
> 
> Something like
> 
> ```
>      if( curl_error( $ch ) || empty( $cont ) ) {
>         error_log( 'Share Count: ' . curl_error($ch) );
>     }
> ```
> 
> The extra empty() statement is if curl itself or the network connection
> barfs and returns an empty string or somesuch.
> 
> I would do the same for any curl statements (here or anywhere else).
> 
> On Tue, Oct 20, 2015 at 11:40 AM, Michael Large notifications@github.com
> wrote:
> 
> > Want to take a look at this?
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/DBSInteractive/DBS-Share-Count/issues/3#issuecomment-149608382
> > .
> 
> ## 
> 
> Hal Burgiss
> CTO and DevOps
> _DBS__>Interactive_ http://www.dbswebsite.com/

## 

Hal Burgiss
CTO and DevOps
_DBS__>Interactive_ http://www.dbswebsite.com/

## 

NOTICE: This E-mail (including attachments) is covered by the Electronic Communications 
Privacy Act, 18 U.S.C. 2510-2521, is confidential and may be legally 
privileged. If you are not the intended recipient, you are hereby notified 
that any retention, dissemination, distribution or copying of this communication 
is prohibited. Please reply to the sender that you have received the 
message in error, then delete it.

**Comment by codethebeard** [comment]
closed in #8

**Comment by mvdmakesthings** [comment]
closed in #8
